### PR TITLE
Incorporate working auth endpoints

### PR DESCRIPTION
**Pull request by maahsnd**
Calls to api on log in and sign up. Valid responses include username and profile photo. Stores these in an authContext, which is then accessed from the navBar to update userIcon. If no profile photo present (it's optional), render the first letter of username. If no username, render a smiley face (should find something better for our anonymous users later on but ok).

### CODE REVIEW

**Comment by DONTSTOPLOVINGMEBABY** [comment]
will check out tomorrow good sir

**Comment by maahsnd** [comment]
Added a login button that is displayed in the navbar if user is not logged in (instead of displaying the userIcon). Refactored userIcon to contain protected links. To that end, created a link for About docs in the navbar, and removed it from the user slide out menu.

File: src/components/authForm/AuthForm.tsx
Comment (DONTSTOPLOVINGMEBABY) [inline src/components/authForm/AuthForm.tsx:1]: Add check for a 409 status code in the conditional in the handleSignupSubmit function. 

```typescript
// ... other conditionals
else if (response.status === HttpStatusCode.Conflict) {
  setError(
    'Username and or email unavailable.'
  )
// continue conditional checks
}
```

**Review by DONTSTOPLOVINGMEBABY — CHANGES_REQUESTED** [review]
This PR is looking really good and is feeling complete. 

There is just one small error that I found in handleSignupSubmit, and I added a comment to it. After this, I think we are set to merge

File: src/components/authForm/AuthForm.tsx
Comment (maahsnd) [reply src/components/authForm/AuthForm.tsx:1]: Good catch. Added in.

**Review by DONTSTOPLOVINGMEBABY — APPROVED** [review]
sweet... see you in like 45 mins

**Pull request merged**